src/v/redpanda/admin_server.cc

```diff
@@ -349,6 +349,39 @@ void admin_server::log_request(
       req.get_url());
 }
 
+void admin_server::log_exception(
+  const ss::sstring& url,
+  const request_auth_result& auth_state,
+  std::exception_ptr eptr) const {
+    using http_status = ss::httpd::reply::status_type;
+    using http_status_ut = std::underlying_type_t<http_status>;
+    const auto log_ex = [&](
+                          std::optional<http_status_ut> status = std::nullopt) {
+        std::stringstream os;
+        const auto username
+          = (auth_state.get_username().size() > 0 ? auth_state.get_username() : "_anonymous");
+        fmt::print(os, "[{}] exception intercepted - url: [{}]", username, url);
+        if (status) {
+            fmt::print(os, " http_return_status[{}]", *status);
+        }
+        fmt::print(os, " reason - {}", eptr);
+        return os.str();
+    };
+
+    try {
+        std::rethrow_exception(eptr);
+    } catch (ss::httpd::base_exception& ex) {
+        const auto status = static_cast<http_status_ut>(ex.status());
+        if (ex.status() == http_status::internal_server_error) {
+            vlog(logger.error, "{}", log_ex(status));
+        } else if (status >= 400) {
+            vlog(logger.warn, "{}", log_ex(status));
```

Comment:
Let’s make sure we do y log any credentials of ppl trying to set passwords for service accounts

Reply:
So not sure if you want to conditional enable logging. Or maybe skip logging depending on path. So paths that set secrets are more discrete about any exceptions 

should this be attached to a higher level handler class that can leverage the same functor but the privacy related ones just log “error on /path” or something generic.

Reply:
Good points, I checked seems like currently the only credentials are the security routes and they are passed via the request body, but your right to future proof this we probably shouldn't be logging the entire URL anyway.

I think its probably OK to just parse everything after the path/ out i.e. remove everything after `?`. The usefulness here is mainly to see the exception.